IHP/IDE/Defaults/TableColumnDefaults.hs

```diff
@@ -0,0 +1,232 @@
+{-|
+Module: Test.IDE.Defaults.TableColumnDefaults
+Copyright: (c) digitally induced GmbH, 2020
+-}
+module IHP.IDE.Defaults.TableColumnDefaults where
+
+import Test.Hspec
+import IHP.Prelude
+import qualified IHP.IDE.CodeGen.ControllerGenerator as ControllerGenerator
+import IHP.ViewPrelude (cs, plain)
+import qualified Text.Megaparsec as Megaparsec
+import IHP.IDE.CodeGen.Types
+import IHP.IDE.SchemaDesigner.Types
+import IHP.NameSupport
+import IHP.IDE.SchemaDesigner.Types
+
+
+{- | Takes a 'Text' value for the name and creates a default column where all values are empty lists, 
+'Nothing' or 'False'; and for 'ColumnType', the default is 'PUUID'.
+
+Add a new field to the 'Column' type in the 
+"IHP.IDE.SchemaDesigner.Types" file and then 
+set its default value here
+
+Defined as:
+
+@
+defColumn :: Column
+defColumn = Column { 
+                 name = ""
+               , columnType = PUUID 
+               , defaultValue = Nothing
+               , notNull = False 
+               , isUnique = False
+               , generator = Nothing
+               }
+@
+
+If you want a different 'PostgresType' you will need to 
+specify with like so by either using a function:
+
+@setColumnType pgt = defColumn {columnType = pgt}@
+
+Or
+
+Just as part of where you're calling it:
+
+@someDefaultColumnType = defColumn {columnType = PDate}@
+-}
+defColumn :: Column
+defColumn = 
+      Column 
+      { name = ""
+      , columnType = PUUID 
+      , defaultValue = Nothing
+      , notNull = False 
+      , isUnique = False
+      , generator = Nothing
+      }
+
+
+{- | Creates an empty table with all values empty. @unlogged@ is set to 'False'.
+
+Defined as such:
+
+@
+emptyTable :: CreateTable
+emptyTable = CreateTable {
+                        name = ""
+                        , columns = []
+                        , primaryKeyConstraint = PrimaryKeyConstraint []
+                        , constraints = []
+                        , unlogged = False
+                        }
+@
+
+-}
+emptyTable :: CreateTable
+emptyTable = CreateTable 
+                  { name = ""
+                  , columns = []
+                  , primaryKeyConstraint = PrimaryKeyConstraint []
+                  , constraints = []
+                  , unlogged = False
+                  }
+
+-- | Takes a name for our table and a list of column and inserts the list
+--   into to our empty table.
+defCreateTable :: Text -> [Column] -> CreateTable
+defCreateTable tablename columns = emptyTable { name = tablename
```

Comment:
It's more for speed: it means you don't have to write the record parts over and over again especially if they're the only two values in `emptyTable` being changed and you could quickly be certain that the only values changed were the name and the columns contained. If other values needed changing, then it's up to the writer to use `emptyTable {..}` or `(defCreateTable tablename columns){..})` -> both options are always available.

Since they were a recurring feature, it made sense to write them this way initially.